backend/pkg/database/apievent.go

```diff
@@ -213,35 +287,54 @@ func APIEventFromDB(event *APIEvent) *models.APIEvent {
 		SpecDiffType:             &event.SpecDiffType,
 		StatusCode:               event.StatusCode,
 		Time:                     event.Time,
+		RequestTime:              event.RequestTime,
 	}
+	for _, ann := range event.Annotations {
+		e.Alerts = append(e.Alerts, &models.ModuleAlert{
+			Alert:      ann.Name,
+			ModuleName: ann.ModuleName,
+			Reason:     string(ann.Annotation),
+		})
+	}
+	return e
 }
 
 func (a *APIEventsTableHandler) CreateAPIEvent(event *APIEvent) {
 	if result := a.tx.Create(event); result.Error != nil {
 		log.Errorf("Failed to create event: %v", result.Error)
 	} else {
-		log.Infof("Event created %+v", event)
+		log.Debugf("Event created %+v", event)
 	}
 }
 
 func (a *APIEventsTableHandler) GetAPIEventsAndTotal(params operations.GetAPIEventsParams) ([]APIEvent, int64, error) {
 	var apiEvents []APIEvent
 	var count int64
 
-	tx := a.setAPIEventsFilters(getAPIEventsParamsToFilters(params), true)
+	tx := a.setAPIEventsFilters(getAPIEventsParamsToFilters(params))
+
 	// get total count item with the current filters
 	if err := tx.Count(&count).Error; err != nil {
 		return nil, 0, err
 	}
-
+	if len(params.AlertIs) > 0 {
+		tx = tx.Joins(fmt.Sprintf("LEFT JOIN %s ea ON %s.%s = ea.%s",
+			eventAnnotationsTableName,
+			apiEventTableName, idColumnName,
+			eventIDColumnName)).
+			Where(fmt.Sprintf("ea.%s IN ?", annotationColumnName), params.AlertIs).
+			Distinct()
```

Comment:
Why distinct is used?

Reply:
because we use join, and this might result in duplicate items